xds/internal/xdsclient/transport/transport.go

```diff
@@ -372,16 +374,32 @@ func (t *Transport) adsRunner(ctx context.Context) {
 // there are new streams) and the appropriate request is sent out.
 func (t *Transport) send(ctx context.Context) {
 	var stream adsStream
+	// The xDS protocol only requires that we send the node proto in the first
+	// discovery request on every stream. Sending the node proto in every
+	// request message wastes CPU resources on the client and the server.
+	sendNodeProto := true
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case stream = <-t.adsStreamCh:
+			// We have a new stream and we've to ensure that the node proto gets
+			// sent out in the first request on the stream. At this point, we
+			// might not have any registered watches. Setting this field to true
+			// here will ensure that the node proto gets sent out along with the
+			// discovery request when the first watch is registered.
+			if len(t.resources) == 0 {
```

Comment:
Consider this scenario. 

1. We have n watches registered and we already have a response from the server. 
    + now sendNodeProto = false 	
3. Now if the stream broke, and when a new stream was created, since t.resources > 0, we would not set sendNodeProto to true. 

When we restart the stream, should we send the node Proto in the first request on the new stream?

Reply:
Looking at the test, it seems like sendNodeProto is being set to true again. So Im guessing the logic checks out then. I'll try to offline this with you. Doesnt seem to be blocking merge.

Reply:
That logic is taken care of by `sendExisting`, which sets a non-empty node proto in the first request. Let me know if that is not apparent, and if a different way of writing the same logic would be more clear.

Reply:
I think calling it like !sendExiting seemed a little confusing to me at first. It doesn't matter for this PR for sure. Please go ahead and merge.